client/src/utils/api.js

```diff
@@ -1,22 +1,43 @@
 // File to contain methods to return data from API calls
 import axios from "axios";
-
+import { BASE_URL } from "./constants";
 /**
  * Get internships with given search options. Returns a promise that resolves to
  * Internship[]
  *
  * @param {*} searchOptions - shape to be determined by backend
  * @return {Promise<Internship[]>}
  */
-export const getSearchInternships = (searchOptions) => {
-  const url = "";
-  const options = {};
-
-  return [
-    // Fill out mock data here
-  ];
+export const getSearchInternships = async (searchOptions) => {
+  const url = BASE_URL;
 
-  // return axios.get(url, options);
+  if (!searchOptions) {
```

Comment:
Instead of two conditions, we could conditionally spread an object

```js
const hasSearchOptions = Object.keys(searchOptions).length

//...

const response = axios.get(
  Url,
  {
    ...(hasSearchOptions
      ? { params: {...searchOptions} }
      : {}
    )
  }
)
```

Hope this looks right; on mobile

Reply:
spread! its a neat pattern i've learned in the last couple years. pretty much, we can use a boolean + ternary to determine if we evaluate and spread the object in the first half of the ternary, or an empty object in the back half of the ternary